### PR TITLE
Set up ability tests ahead of locking down courses to audiences

### PR DESCRIPTION
**Pull request by dmcavoy**
Improves the testing in ability_test.rb in order to prepare for locking down courses to audiences. Adds more coverage of the current behavior. This will allow it to be very clear what has changed in access when we lock down courses. 

## Links

[Product Spec](https://docs.google.com/document/d/1qZpc90daxQiiqpcQUiUiGV2tQoaYEQUzOA72UZzBlVE/edit#)
[Eng Plan](https://docs.google.com/document/d/1V61xONU0-mleMEEdHWNxhvZtYIPRCB31mT_hbUXusCQ/edit#heading=h.n9isd44okbq7)

## Testing story

- This PR is all about adding more test coverage of the current behavior

### CODE REVIEW

File: dashboard/test/models/ability_test.rb [hunk 1]
```diff
@@ -3,17 +3,35 @@
 class AbilityTest < ActiveSupport::TestCase
   self.use_transactional_test_case = true
   setup_all do
-    @public_unit_group = create(:unit_group)
-    @public_script = create(:script).tap do |script|
-      @public_script_level = create(:script_level, script: script)
+    @public_teacher_to_student_unit_group = create(:unit_group)
```
Comment (bethanyaconnor) [inline dashboard/test/models/ability_test.rb]: if the audiences are important, can you set them explicitly here?

File: dashboard/test/models/ability_test.rb [hunk 2]
```diff
@@ -29,70 +47,302 @@ class AbilityTest < ActiveSupport::TestCase
...
+    refute ability.can?(:destroy, Level)
+    refute ability.can?(:destroy, Activity)
     refute ability.can?(:read, Section)
+    assert ability.can?(:read, Script.find_by_name('ECSPD'))
+    assert ability.can?(:read, Script.find_by_name('flappy'))
 
     assert ability.can?(:read, Script.find_by_name('ECSPD'))
     assert ability.can?(:read, Script.find_by_name('flappy'))
 
     refute ability.can?(:read, @in_development_script)
```
Comment (bethanyaconnor) [inline dashboard/test/models/ability_test.rb:164]: nit: this check isn't in the plc pilot tests but might be worth adding to those as well?

**Review by bethanyaconnor — APPROVED** [review]
Thanks for adding all these tests!

File: dashboard/test/models/ability_test.rb (hunk 2 above)
Comment (dmcavoy) [reply dashboard/test/models/ability_test.rb:164]: I'm not sure I understand what test you are suggesting I add this too. Could you clarify?

Comment (bethanyaconnor) [reply dashboard/test/models/ability_test.rb:164]: I was thinking of "as pilot plc reviewer" and "as pilot facilitator". It's not a big deal, just wondering if we should test for this in all tests

Comment (dmcavoy) [reply dashboard/test/models/ability_test.rb:164]: I see these lines being tested in those test right now.

Comment (bethanyaconnor) [reply dashboard/test/models/ability_test.rb:164]: oh weird, maybe the diff hid them. Anyway, this PR is def good to go!

**Pull request merged**